telebot/async_telebot.py

```diff
@@ -446,9 +452,10 @@ async def _process_polling(self, non_stop: bool=False, interval: int=0, timeout:
                     if not handled:
                         logger.error('Unhandled exception (full traceback for debug level): %s', self.__hide_token(str(e)))
                         logger.debug(self.__hide_token(traceback.format_exc()))
+                        error_interval = await self._handle_error_interval(error_interval)
                         
                     if non_stop or handled:
-                        await asyncio.sleep(2)
+                        #await asyncio.sleep(2) # used error_interval instead
```

Comment:
Changed behaviour!

In old version sleep(2) is for non_stop or **handled**. In new version it is for **not hadled** and **independently from non_stop**.

1. Adding for not handled is fine.
2. Removing from handled **is not fine**, because handled exceptions will cause now immediate re-run instead of 2 second delay.
3. Make sleep() for non_stop looks also not needed.

I propose:

```
                    if not handled:
                        logger.error('Unhandled exception (full traceback for debug level): %s', self.__hide_token(str(e)))
                        logger.debug(self.__hide_token(traceback.format_exc()))

                    if non_stop:
                        error_interval = await self._handle_error_interval(error_interval)

```

Reply:
so we still don't need sleep for handled? or should I add sleep to handled (2 seconds) back?

Reply:
We need to sleep both for handled (as before) and not handled (it's reasonable).

But I propose only if non_stop: otherwise sleeping is useless, because polling will stop anyway.

Reply:
wait, before, script slept 2 seconds for handled or non_stop. So, should I:
- Change your last condition to "if non_stop or handled"?
- Add another condition for handled, making the script sleep for 2 seconds?

Reply:
this will not affect not handled: there will be no sleeping as you said, the script will be useless. currently we are talking about handled. Or, do you think the user should manually do that and not rely on internal behaviour?

Reply:
I am talking about the need to have "if handled" condition. 
In previous version, as you know, sleep is triggered on handled or non_stop. After I copy your suggestion, sleep is not triggered for handled. is that fine?

Reply:
> After I copy your suggestion, sleep is not triggered for handled. is that fine?

It is not. But if you copy **my version** - it will work both for handled and not handled.

Reply:
> After I copy your suggestion, sleep is not triggered for handled. is that fine?

It is not. 

I don't know what you did, but if you'll take **my version** - it will work both for handled and not handled. Isn't it?

Reply:
ah, sorry 
was sleepy a little, should work, yep